sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregation/MinMaxSumCountAggregation.java

```diff
@@ -18,40 +18,33 @@
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
-final class MinMaxSumCountAggregation implements Aggregation {
+final class MinMaxSumCountAggregation extends AbstractAggregation<MinMaxSumCountAccumulation> {
   static final MinMaxSumCountAggregation LONG_INSTANCE =
       new MinMaxSumCountAggregation(LongMinMaxSumCountAggregator.getFactory());
   static final MinMaxSumCountAggregation DOUBLE_INSTANCE =
       new MinMaxSumCountAggregation(DoubleMinMaxSumCountAggregator.getFactory());
 
-  private final AggregatorFactory<?> aggregatorFactory;
-
-  private MinMaxSumCountAggregation(AggregatorFactory<?> aggregatorFactory) {
-    this.aggregatorFactory = aggregatorFactory;
-  }
-
-  @Override
-  public AggregatorFactory<?> getAggregatorFactory() {
-    return aggregatorFactory;
+  private MinMaxSumCountAggregation(
+      AggregatorFactory<MinMaxSumCountAccumulation> aggregatorFactory) {
+    super(aggregatorFactory);
   }
 
   @Override
-  public Accumulation merge(Accumulation a1, Accumulation a2) {
-    MinMaxSumCountAccumulation minMaxSumCountAccumulation1 = (MinMaxSumCountAccumulation) a1;
-    MinMaxSumCountAccumulation minMaxSumCountAccumulation2 = (MinMaxSumCountAccumulation) a2;
+  public MinMaxSumCountAccumulation merge(
+      MinMaxSumCountAccumulation a1, MinMaxSumCountAccumulation a2) {
     return MinMaxSumCountAccumulation.create(
-        minMaxSumCountAccumulation1.getCount() + minMaxSumCountAccumulation2.getCount(),
-        minMaxSumCountAccumulation1.getSum() + minMaxSumCountAccumulation2.getSum(),
-        Math.min(minMaxSumCountAccumulation1.getMin(), minMaxSumCountAccumulation2.getMin()),
-        Math.max(minMaxSumCountAccumulation1.getMax(), minMaxSumCountAccumulation2.getMax()));
+        a1.getCount() + a2.getCount(),
+        a1.getSum() + a2.getSum(),
+        Math.min(a1.getMin(), a2.getMin()),
+        Math.max(a1.getMax(), a2.getMax()));
   }
 
   @Override
   public MetricData toMetricData(
       Resource resource,
       InstrumentationLibraryInfo instrumentationLibraryInfo,
       InstrumentDescriptor descriptor,
-      Map<Labels, Accumulation> accumulationMap,
+      Map<Labels, ? extends Accumulation> accumulationMap,
```

Comment:
for future, I might name this parameter "accumulationsByLabels". No need to change it now.